app/components/hook-switch.js

```diff
@@ -13,12 +18,12 @@ export default Ember.Component.extend({
       let pusher = this.get('pusher'),
         repoId = hook.get('id');
 
-      return hook.toggle().then(() => {
+      yield hook.toggle().then(() => {
         pusher.subscribe(`repo-${repoId}`);
       }, () => {
         this.toggleProperty('hook.active');
         return this.sendAction('onToggleError', hook);
       });
     }
-  }
+  }),
```

Comment:
I don't think so. The task will fire whenever a user clicks to enable/disable any hook, and we want that behavior isolated. If we `drop` or `restart`, wouldn't that mean we'd potentially impact a previous toggle, e.g. if a user clicks two different hooks in quick succession?